libr/anal/fcn.c

```diff
@@ -264,76 +264,55 @@ static ut64 search_reg_val(RAnal *anal, ut8 *buf, ut64 len, ut64 addr, char *reg
 #define gotoBeach(x) ret=x;goto beach;
 #define gotoBeachRet() goto beach;
 
-R_API void fill_args (RAnal *anal, RAnalFunction *fcn, RAnalOp *op) {
-	char* varname;
-	switch (op->stackop) {
-	case R_ANAL_STACK_NULL:
-	case R_ANAL_STACK_NOP:
-	case R_ANAL_STACK_ALIGN:
-	case R_ANAL_STACK_INC: {
-		char bp[50]={',',0};
-		strcat (bp, anal->reg->name[R_REG_NAME_BP]);
-		strcat (bp, ",+,[4],");
-		//TODO char, short, and long in case of 64bit it wont be [4]
-		//This will be some hints for type propagation imho
-		char *esil_buf, *ptr_end, *addr, *op_esil;
-		st64 ptr;
-		op_esil = r_strbuf_get (&op->esil);
-		if (!op_esil) {
-			return;
-		}
-		esil_buf = strdup (op_esil);
-		if (!esil_buf) {
-			return;
-		}
-		ptr_end = strstr (esil_buf, bp);
-		if (!ptr_end) {
-			free (esil_buf);
-			break;
-		}
-		*ptr_end = 0;
-		addr = ptr_end;
-		while ((*addr != '0' || *(addr+1) != 'x') &&
-			addr >= esil_buf && *addr != ',' ) {
-			addr--;
-		}
-		if (strncmp (addr, "0x", 2)) {
-			free (esil_buf);
-			break;
-		}
-		ptr = (st64)r_num_get (NULL, addr);
-		varname = get_varname (anal, ARGPREFIX, R_ABS (ptr));
-		r_anal_var_add (anal, fcn->addr, 1, ptr, fcn->call, NULL, anal->bits/8, varname);
-		r_anal_var_access (anal, fcn->addr, fcn->call, 1, ptr, 0, op->addr);
+void extract_arg (RAnal *anal, RAnalFunction *fcn, RAnalOp *op, char *reg, char *sign, char type) {
+	char *varname, *esil_buf, *ptr_end, *addr, *op_esil;
+	st64 ptr;
+	char sig[50] = {',', 0};
```

Comment:
char foo[50] = “,”; looks more readable to me

> On 04 May 2016, at 15:33, Ahmed Mohamed Abd El-MAwgood <notifications@github.com> wrote:
> 
> In libr/anal/fcn.c <https://github.com/radare/radare2/pull/4759#discussion_r62038750>:
> 
> > -		while ((*addr != '0' || *(addr+1) != 'x') &&
> > -			addr >= esil_buf && *addr != ',' ) {
> > -			addr--;
> > -		}
> > -		if (strncmp (addr, "0x", 2)) {
> > -			free (esil_buf);
> > -			break;
> > -		}
> > -		ptr = (st64)r_num_get (NULL, addr);
> > -		varname = get_varname (anal, ARGPREFIX, R_ABS (ptr));
> > -		r_anal_var_add (anal, fcn->addr, 1, ptr, fcn->call, NULL, anal->bits/8, varname);
> > -		r_anal_var_access (anal, fcn->addr, fcn->call, 1, ptr, 0, op->addr);
> > +void extract_arg (RAnal *anal, RAnalFunction *fcn, RAnalOp *op, char *reg, char *sign, char type) {
> > +	char *varname, *esil_buf, *ptr_end, *addr, *op_esil;
> > +	st64 ptr;
> > +	char sig[50] = {',', 0};
> constructing string that strarts with a colon
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub <https://github.com/radare/radare2/pull/4759/files/232e658d4a9a5e72c7c74d0d80f1b1376c6d6e20#r62038750>